Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/PressableButtons/PressableButton.cs

```diff
@@ -25,29 +26,72 @@ public class PressableButton : MonoBehaviour, IMixedRealityTouchHandler
 
         [SerializeField]
         [Header("Press Settings")]
+        [Tooltip("The offset at which pushing starts.")]
+        private float startPushDistance = 0.0f;
+        public float StartPushDistance { get => startPushDistance; set => startPushDistance = value; }
+
+        [SerializeField]
         [Tooltip("Maximum push distance")]
         private float maxPushDistance = 0.2f;
+        public float MaxPushDistance { get => maxPushDistance; set => maxPushDistance = value; }
 
         [SerializeField]
         [FormerlySerializedAs("minPressDepth")]
         [Tooltip("Distance the button must be pushed until it is considered pressed.")]
         private float pressDistance = 0.02f;
+        public float PressDistance { get => pressDistance; set => pressDistance = value; }
 
         [SerializeField]
         [FormerlySerializedAs("withdrawActivationAmount")]
         [Tooltip("Withdraw amount needed to transition from Pressed to Released.")]
         private float releaseDistanceDelta = 0.01f;
+        public float ReleaseDistanceDelta { get => releaseDistanceDelta; set => releaseDistanceDelta = value; }
 
         [SerializeField]
-        [Tooltip("Speed of the object movement on release.")]
-        private float returnRate = 25.0f;
+        [Tooltip("Speed for retracting the moving button visuals on release.")]
+        [FormerlySerializedAs("returnRate")]
+        private float returnSpeed = 25.0f;
 
         [SerializeField]
         [Tooltip("Ensures that the button can only be pushed from the front. Touching the button from the back or side is prevented.")]
         private bool enforceFrontPush = true;
 
-        [Header("Position markers")]
-        [Tooltip("Used to mark where button movement begins. If null, it will be automatically generated.")]
+        public enum SpaceMode
+        {
+            World,
+            Local
+        }
+
+        [SerializeField]
+        [HideInInspector]
+        private SpaceMode distanceSpaceMode = SpaceMode.World;
+
+        public SpaceMode DistanceSpaceMode
+        {
+            get => distanceSpaceMode;
+            set
+            {
+                // Convert world to local distances and vice versa whenever we switch the mode
+                if (value != distanceSpaceMode)
+                {
+                    distanceSpaceMode = value;
+                    float scale;
+                    if (distanceSpaceMode == SpaceMode.Local)
+                    {
+                        scale = transform.InverseTransformVector(WorldSpacePressDirection).magnitude;
+                    }
+                    else
+                    {
+                        scale = transform.TransformVector(WorldSpacePressDirection).magnitude;
```

Comment:
Do you really intend to scale the **world**space press direction here? Seems like you are scaling a world space vector by a world space transform.

Reply:
should be fixed